### DISCUSSION TITLE
Name of "Chunk": Chunks/Knowledge Fragments/etc

### DISCUSSION THREAD

**Discussion by balacij** [JacquesCarette/Drasil#3017] [General]
_Background_: In #2911, we briefly discussed an alternative name to "chunks" (i.e., we noted "knowledge fragments"), but I didn't want to continue the discussion on that ticket, a "Discussion" might be more appropriate.

_Disclaimers_: 
i) I am not sure if "chunks" was named as such with "chunking"/"chunks" (in the psychological sense) was taken into consideration when deciding on a name for "chunks."
ii) I'm also not sure if I'm allowed to quote a whole page or two from the "Sense of Style" book, so I will just share the book with anyone IRL if they need it, or, if it's allowed, I can post it here.
iii) We also very briefly discussed this in the last meeting (#3013).

"Sense of Style"s third chapter briefly discusses "chunking" ([wikipedia](https://en.wikipedia.org/wiki/Chunking_(psychology)) and [cited paper](https://www.musanim.com/miller1956/)) from a psychological point of view, where a "chunk" and the process of forming chunks, called "chunking", is defined and commonly known. A "chunk", from what I've understood, is essentially any fragment of information that is split up in some way that we can remember. A common example of a "chunk" is splitting up some phone number into 3 parts so that it can be remembered with a rhythm/tune (e.g., 9051112222 might be chunked as 905, 3 "1"s, 4 "2"s), or even mnemonic tricks to remember things (such as "SOHCAHTOA" to remember how to define trigonometric functions in terms of the sides of a triangle).

On page 68, the paragraph which starts off with "Chunking is not just a trick for improving memory; it's the lifeblood of higher intelligence" (I'm not quoting this in logical fallacy, but to help with locating the paragraph in the book), discusses, through example, how the meaningful chunks are abstractions over absolute semantics. Largely, I believe this coincides with how we chunks are built as well (#2883).

Since it seems that the field of psychology already has this term defined and agreed upon, it seems that continuing to call "chunks" chunks is appropriate. "Chunking" would also be a good term to adopt, assuming we don't already use it (I don't recall any references).

**Comment by smiths** [comment]
@balacij as we briefly discussed in #3013, our earlier use of the term chunk was inspired by Knuth's [Literate Programming](https://en.wikipedia.org/wiki/Literate_programming).  Knuth divides the code into chunks within the literate document.  The chunks are presented in the order, and at the abstraction level, appropriate for communicating to a human being.  In our meeting I incorrectly said that chunks were also used for chunks of documentation.  From reviewing the Wikipedia page, that isn't actually true.  Knuth reserved the term chunk for code and lower-level macros.

Given that Knuth's use of the term didn't entirely match our use of the term, and that chunk wasn't rigorously defined in our case, we started to get away from using the term.  It works in a "hand-wavy" sense as a short-hand for a piece of knowledge, but it didn't feel like it was the right term as our work matured.

@balacij, after our group meeting and after reading your discussion, I'm much more in favour of the term "chunk."  Knowing that the term is used by psychologists to refer to a piece of information makes it seem well suited to our purpose.  The idea that we call decomposing a bigger piece of information "chunking" also sounds good to me.

**Comment by JacquesCarette** [comment]
My discomfort with "chunk" stemmed entirely from the huge drift away from Knuth's original conception. However, the use in psychology, well described above by @balacij , reconciles me with using that word.

Thanks!

**Reply by balacij** [reply]
I'm sure @smiths will be happy to continue "chunk" usage :smile:. At times, it might also be beneficial to sometimes call chunks "fragments" when we want to indicate that the chunk is a piece of another whole.

**Comment by peter-michalski** [comment]
> A "chunk", from what I've understood, is essentially any fragment of information that is split up in some way that we can remember.

The vagueness of the concept when described like this, in an abstract sense, is great. It's a nice term for describing a high level concept used in Drasil to categorize rather than remember. I feel that this can create confusion when the concept is applied at a lower level - different types of chunks should then become very well defined. Of course many types of chunks are identified in Drasil, somewhat mitigating confusion. Now, is there room for improvement when dealing with / identifying (naming) possible types of chunks (and their relationships) in Drasil?

**Reply by balacij** [reply]
> Now, is there room for improvement when dealing with / identifying (naming) possible types of chunks (and their relationships) in Drasil?

This part is indeed interesting! We almost want to find a structure to things. Did you have something in mind regarding this?

**Reply by peter-michalski** [reply]
> Did you have something in mind regarding this?

I don't have anything specific in mind, just wondering if there is an ideal chunk structure for scs generation (maybe Drasil already has it?). Furthermore, could there be a methodology for identifying suitable chunk structures for subdomains or even specific projects. A first step would naturally be to carefully consider and parse possible information in each system/domain.

**Reply by balacij** [reply]
Hmm, I don't really know. We would need to define what "ideal" means w.r.t. the structures, which would likely have multiple facets, and create fit criteria. The questions and solutions would also always be susceptible to our personal biases and knowledge, so I imagine any questions and solutions we come up with would change over time.

**Reply by peter-michalski** [reply]
You're right. I think the word "ideal" is not appropriate. It should be a chunk structure that fits certain criteria - which could also improve over time.
